docs/docs/create-transformer-plugin.md

```diff
@@ -2,36 +2,272 @@
+
+## Transformer plugin example
 
-This loose coupling between the data source and transformer plugins allow Gatsby
-site builders to quickly assemble complex data transformation pipelines with
-little work.
+The [`gatsby-transformer-remark`](/packages/gatsby-transformer-remark/) plugin "transforms" markdown nodes (fetched for example from the filesystem source plugin) into html and provides additional node details such as for example the `timeToRead` the `tableOfContents` and the `excerpt`.
 
 ## What does the code look like?
 
-Just like a source plugin, a transformer plugin is a normal NPM package. It has a `package.json` file with optional
-dependencies as well as a `gatsby-node.js` file where you implement Gatsby's Node.js
-APIs. Transformer plugins make use of the `setFieldsOnGraphQLNodeType` function and they commonly provide this functionality in the `extend-node-type` file.
+Just like a source plugin, a transformer plugin is a normal NPM package. It has a `package.json` file with optional dependencies as well as a `gatsby-node.js` file where you implement Gatsby's Node.js APIs.
+
+`gatsby-transformer-yaml` is transformer plugin that looks for new nodes with a media type of text/yaml (e.g. a .yaml file) and creates new YAML child node(s) by parsing the YAML source into JavaScript objects.
+
+As an example, let's partially rebuild a simplified `gatsby-transformer-yaml` directly in an example site. Let's say we have a default gatsby starter site which includes a YAML file:
+
+```yaml:title=src/data/example.yml
+- name: Jane Doe
+  bio: Developer based in Somewhere, USA
+- name: John Smith
+  bio: Developer based in Maintown, USA
+```
+
+### Make sure the data is sourced
 
-Your `gatsby-node.js` should look something like:
+First, we make sure the file is sourced, using `gatsby-source-filesystem` in `gatsby-config.js`:
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        path: `./src/data/`,
+      },
+    },
+  ],
+}
+```
+
+Now if we query `allFile`...
+
+```graphql
+query {
+  allFile {
+    edges {
+      node {
+        internal {
+          type
+          mediaType
+          description
+          owner
+        }
+      }
+    }
+  }
+}
+```
+
+... We see we have the sourced file.
+
+```json
+{
+  "data": {
+    "allFile": {
+      "edges": [
+        {
+          "node": {
+            "internal": {
+              "contentDigest": "c1644b03f380bc5508456ce91faf0c08",
+              "type": "File",
+              "mediaType": "text/yaml",
+              "description": "File \"src/data/example.yml\"",
+              "owner": "gatsby-source-filesystem"
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### Transform nodes of type `text/yaml`
+
+Now we can transform the sourced data by hooking into the `onCreateNode` API in `gatsby-node.js`.
```

Comment:
instead of "sourced data" — use Gatsby terminology "transform the newly created `File` nodes"